src/ffi-buffer-clisp.lisp

```diff
@@ -58,40 +89,55 @@
   (setf (ffi:memory-as (clisp-ffi-buffer-pointer buf) 'ffi:uint8 index) val))
 (defsetf buffer-elt set-buffer-elt)
 
-(declaim
- (inline calc-buf-end))
-
-;; to calculate non NIL value of the buffer end index
-(defun calc-buf-end (buf-start seq seq-start seq-end)
-  (+ buf-start
-     (- (or seq-end (length seq))
-        seq-start)))
+(defparameter *mem-max* 1024 "so *-REPLACE require the expected O(1) memory")
```

Comment:
The `*mem-max*` is intended to limit the allocations done by `memory-as`, right?  Question, if we copy 2048 bytes, calling memory-as twice for 1024 bytes, is it really better than a single call for 2048 bytes? The total amount of allocated memory is the same. If it is garbage collected, collecting two objects may be more work for GC than one object. Is it really beneficial?

Also, in context of cl+ssl, the maximum size of arrays copied with `s/b-replace`, `b/s-replace` is limited by the buffer size, which defaults to 2048 bytes. Not a big difference from the 1024 `*mem-max*` here.

Are you thinking of the `*-replace` functions as more general purpose utilities, that must be prepared to any size of collections copied?